### PR TITLE
feat(build-package): Include draft in roots and manifest versions

### PR DESCRIPTION
**Pull request by spencerwilson**
- Update generator impls for all targets to incorporate a `Prerelease` component (assumed to be `draft.N`, if present) into the root dirs and package instance versions

### CODE REVIEW

**Comment by spencerwilson** [comment]
Current dependencies on/for this PR:
* main
  * **PR #144** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/144" target="_blank">Graphite</a> 
    * **PR #145** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/145" target="_blank">Graphite</a> 
      * **PR #150** <a href="https://app.graphite.dev/github/pr/patch-tech/dpm/150" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/patch-tech/dpm/150?utm_source=stack-comment).

File: src/codegen/csharp.rs [hunk 1]
```diff
@@ -111,6 +112,26 @@ namespace Dpm \\{
 }
 ";
 
+/// Returns a version string for a Node.js package instance:
```
Comment (ajmasci) [inline src/codegen/csharp.rs:115]: typo: `Node.js` -> `C#`

File: src/codegen/python.rs [hunk 2]
```diff
@@ -164,6 +165,23 @@ static VERSION_TEMPLATE: &str = "
...
+/// Returns a version string for a Python package instance:
+///   dataset-version "." code-version ("a" draft-number)?
+/// See: https://peps.python.org/pep-0440/#public-version-identifiers
+fn package_instance_version(v: &Version) -> String {
+    if v.pre.is_empty() {
+        format!("{}.{}", v, PYTHON_VERSION)
+    } else {
+        // Assume this has form "draft.<number>", and so can be
+        // joined with the rest of the string via a "a".
+        let (_, draft_number) = v.pre.split_at(v.pre.find('.').unwrap());
```
Comment (ajmasci) [inline src/codegen/python.rs:177]: Would it be a good idea to assert that `v.pre` has the expected form before doing the split? How would this code fail if the expected form is not present?

File: src/codegen/csharp.rs [hunk 3]
```diff
@@ -361,7 +381,7 @@ impl Generator for Csharp<'_> {
     fn manifest(&self) -> Manifest {
         let dp = self.data_package();
         let pkg_name: String = self.package_name(&dp.package_name);
-        let version = format!("{}-{}", dp.version.version, CSHARP_VERSION);
+        let version = package_instance_version(&dp.version.version);
 
         let src_dir = self.source_dir();
         let proj_file_name = format!("{pkg_name}.csproj");
```
Comment (ajmasci) [inline src/codegen/csharp.rs:387]: <i class='graphite__hidden'>[Re: lines 396 to 396]<br/></i>

I verified that using the following works to set the correct version in the built package
```xml
<Version>{version}</Version>
```

**Review by ajmasci — APPROVED** [review]

File: src/codegen/python.rs (hunk 2 above)
Comment (ajmasci) [reply src/codegen/python.rs:177]: Verified that the backend always returns a `draft.<number>` for the pre-release version.

**Pull request merged**